pytest.ini

```diff
@@ -4,6 +4,8 @@
 [pytest]
 # In general, all warnings are treated as errors. Here are the exceptions:
 #   1- decodestring: https://github.com/rthalley/dnspython/issues/338
+#   2- Python 2 deprecation: https://github.com/certbot/certbot/issues/8388
+#      (to be removed with Certbot 1.12.0 and its drop of Python 2 support)
```

Comment:
https://semver.org/ If certbot follows Semantic Version, shouldn't this be 2.0.0?

Reply:
I think it's debatable. From that page:

> What should I do if I update my own dependencies without changing the public API?
>
> That would be considered compatible since it does not affect the public API. Software that explicitly depends on the same dependencies as your package should have their own dependency specifications and the author will notice any conflicts. Determining whether the change is a patch level or minor level modification depends on whether you updated your dependencies in order to fix a bug or introduce new functionality. I would usually expect additional code for the latter instance, in which case it’s obviously a minor level increment.

In my experience, some Python projects bump the major version when dropping support for a version of Python and some do not. We have never done it ourselves in the past and no one ever complained about it. We were not planning on updating the major version now and I still don't think we should.

I think it causes more harm than good since it will cause downstream packagers/users to worry about API changes when they're aren't any. If someone tries to install Certbot on Python 2 (with semi-updated versions of `setuptools`/`pip`), it will either install the last version that worked with Python 2 or refuse to install it saying Python 2 isn't supported. Us bumping the major version does not help them in my opinion.